ladybug_comfort/collection/_base.py

```diff
@@ -0,0 +1,65 @@
+# coding=utf-8
+"""Comfort data collection base object."""
+
+from ladybug._datacollectionbase import BaseCollection
+
+
+class ComfortCollection(object):
+    """Thermal comfort datacollection base class."""
+    _model = None
+
+    def __init__(self):
+        self._calc_length = 0
+        self._base_collection = None
+        self._input_collections = []
+
+    @property
+    def comfort_model(self):
+        """Return the name of the model to which the comfort datacollection belongs."""
+        return self._model
+
+    @property
+    def calc_length(self):
+        """The number of values in the Data Collections of this object."""
+        return self._calc_length
+
+    def _check_datacoll(self, data_coll, dat_type, unit, name):
+        """Check the data type and units of a Data Collection."""
+        assert isinstance(data_coll, BaseCollection), '{} must be a ' \
+            'Data Collection. Got {}.'.format(name, type(data_coll))
+        assert isinstance(data_coll.header.data_type, dat_type) and \
+            data_coll.header.unit == unit, '{} must be {} in {}. ' \
+            'Got {} in {}'.format(name, dat_type().name, unit,
+                                  data_coll.header.data_type.name,
+                                  data_coll.header.unit)
+        self._input_collections.append(data_coll)
```

Comment:
I think you are misunderstanding the point of this hidden `_input_collections` attribute.  The reason why it is there is so that we can run a check to make sure that all collections that were input to the object are aligned with one another as you can see [here](https://github.com/chriswmackey/ladybug-comfort/blob/master/ladybug_comfort/collection/utci.py#L100).  This way, we don't have to run this alignment check on an input that is a single float number (only if it is a data collection that has passed through this check that is specifically for data collections).  Do you want me to change the name of the `_input_collections` attribute to `_input_collections_to_check_for_alignment` or something like that?  It seems like that would be a long name for a hidden attribute, though.

Reply:
The other way around. `check_datacoll` should be used for checking the data collections and appending the data can happen after that. Something like this:

 ```python
    def _check_input(self, data_coll, dat_type, unit, name):
        """Check the a Data Collection."""
        if isinstance(data_coll, BaseCollection):
            # check data collection
            self._check_datacoll(data_coll, dat_type, unit, name)
            # append it to input collections
            self._input_collections.append(data_coll)
            return dc
        else:
            # the rest of the code
```

Reply:
Ok, I can do it like this, though there will be several other places in the code where I need to append a data collection the `_input_collections`.  I will admit that I don't totally understand the reason for why it should be that way given that there are no situations on the comfort collections where you want to check a data collection without seeing that it is also aligned with other collections.  You are the one looking at the code with the fresh eyes, though, so I trust your judgement that this current implementation is confusing.

Reply:
Actually, I can see why it is confusing.  I think I was too deep into the revisions that gave rise to that method to see it.  I will change it now.